src/Shouldly.Tests/ShouldBe/EnumerableType/EnumerableOfReadOnlyMemoryScenario.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Shouldly.Tests.Strings;
+using Xunit;
+
+namespace Shouldly.Tests.ShouldBe.EnumerableType
+{
+    public class EnumerableOfReadOnlyMemoryScenario
+    {
+        [Fact]
+        public void EnumerableOfReadOnlyMemoryScenarioShouldFail()
+        {
+            ReadOnlyMemory<byte> foo = new byte[] { 1, 2, 3 }.AsMemory();
+            ReadOnlyMemory<byte> bar = new byte[] { 1, 2 }.AsMemory();
+
+            Verify.ShouldFail(() =>
+                    foo.ShouldBe(bar, "Some additional context"),
+
+errorWithSource:
+@"foo
+    should be
+System.ReadOnlyMemory<Byte>[2]
+    but was
+System.ReadOnlyMemory<Byte>[3]
```

Comment:
Normally the elements are listed in the error message. Could you see what it would take to enumerate here too?

Reply:
The method that generates the error method takes in the original and expected objects and passes it to this method - https://github.com/shouldly/shouldly/blob/5fc845c32ccbf2b7e8a6c9186d3cd6382d9ae430/src/Shouldly/Internals/ShouldlyCoreExtensions.cs#L22

We can convert actual and expected `Memory` items into Enumerables in
1.  `ShouldBe()` and work with Enumerables
2.  this line while constructing `ExpectedActualShouldlyMessage` https://github.com/shouldly/shouldly/blob/5fc845c32ccbf2b7e8a6c9186d3cd6382d9ae430/src/Shouldly/Internals/ShouldlyCoreExtensions.cs#L39

Reply:
Thanks, that helps. It looks like those values are used here:

https://github.com/shouldly/shouldly/blob/5fc845c32ccbf2b7e8a6c9186d3cd6382d9ae430/src/Shouldly/MessageGenerators/ShouldBeMessageGenerator.cs#L20-L21

And the string is built here:

https://github.com/shouldly/shouldly/blob/5fc845c32ccbf2b7e8a6c9186d3cd6382d9ae430/src/Shouldly/Internals/StringHelpers.cs#L42-L51

I'm thinking we should obtain the IEnumerable in ToStringAwesomely for consistency because it sometimes uses the type full name and ToString of the original instance, and we wouldn't want to use the instance returned from MemoryMarshal.ToEnumerable for that.

Reply:
Pushed a change